src/common/transformations/src/transformations/mlir/mlir_op.cpp

```diff
@@ -337,6 +337,15 @@ MLIREvaluateGcGPU::MLIREvaluateGcGPU(OwningOpRef<mlir::ModuleOp> _module, std::s
 
     gc::gpu::OclModuleBuilderOpts opts;
     OPENVINO_MLIR_DEBUG(opts.printIr = true);
+#ifdef MLIR_RUNNER_UTILS_PATH
+    #define STRINGIFY(x) #x
+    #define TOSTRING(x) STRINGIFY(x)
+    #define MLIR_RUNNER_UTILS_SO TOSTRING(MLIR_RUNNER_UTILS_PATH) "/libmlir_runner_utils.so"
+    #define MLIR_C_RUNNER_UTILS_SO TOSTRING(MLIR_RUNNER_UTILS_PATH) "/libmlir_c_runner_utils.so"
+    SmallVector<StringRef, 2> shared_libs({MLIR_RUNNER_UTILS_SO, MLIR_C_RUNNER_UTILS_SO});
```

Comment:
This seems not relocatable. I think it could use something similar to https://github.com/openvinotoolkit/openvino/blob/master/src/common/util/include/openvino/util/shared_object.hpp. There's a library search part in the implementation, e.g., https://github.com/openvinotoolkit/openvino/blob/f0da7075169b97f6523d8f465cbb6ab76f995324/src/common/util/src/os/lin/lin_shared_object_loader.cpp#L16.